samples/metairq_dispatch/src/msgdev.c

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <timeout_q.h>
+#include "main.h"
+
+/* This file implements a fake device that creates and enqueues
+ * "struct msg" messages for handling by the rest of the test.  It's
+ * based on Zephyr kernel timeouts only.
+ */
+
+/* Note: we use the internal timeout API to get tick precision,
+ * k_timer limits us to milliseconds.
+ */
+static struct _timeout timeout;
+
+/* The "proc_cyc" parameter in the message, indicating how many cycles
+ * the target thread should delay while "processing" the message, will
+ * be a random number between zero and this value.
+ */
+u32_t max_duty_cyc;
+
+u32_t msg_seq;
+
+K_MSGQ_DEFINE(hw_msgs, sizeof(struct msg), 2, sizeof(u32_t));
+
+static void timeout_reset(void);
+
+/* Use a custom RNG for good statistics, sys_rand32_get() is just a
+ * timer counter on some platforms.  Note that this is used only
+ * inside the ISR and needs no locking for the otherwise non-atomic
+ * state.
+ */
+static u32_t rand32(void)
```

Comment:
mixed feeling about it ... It is just masking another problem.

Reply:
I dunno, this is a sample.  It's only going to be useful to people to inspect and emulate if it works correctly.  And right now it doesn't work correctly with sys_rand32_get().  I agree that's a bug and it should be fixed, but committing a broken sample or delaying merge while we wait for a bug fix to an unrelated subsystem seems like not a great choice either.

Reply:
agreed

Reply:
I disagree about it just be a sample, I really believe that people start applications copying from samples. I have to be very careful about our samples to have there best practices from Zephyr development and not only about the features that is being shown.

But, I don't want to pedantic with this particular pr, since my comment apply for all of them :)
The sample is correct and is addressing that feature.